migrations/back_populate_uuids_plate_barcodes.py

```diff
@@ -0,0 +1,202 @@
+
+    Returns:
+        Nothing
+    """
+    # counters to track progress
+    counter_mongo_update_successes = 0
+    counter_mongo_update_failures = 0
+
+    with create_mongo_client(config) as client:
+        mongo_db = get_mongo_db(config, client)
+        samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
+
+        check_samples_are_valid(config, samples_collection, source_plate_barcodes)
+
+        for source_plate_barcode in source_plate_barcodes:
+            LOGGER.info(f"Processing source plate barcode {source_plate_barcode}")
+
+            # List[SampleDoc]
+            sample_docs = get_mongo_samples_for_source_plate(samples_collection, source_plate_barcode)
+
+            # iterate through samples
+            for sample_doc in sample_docs:
+                # will every sample doc have a plate_barcode and lab id?
+                LOGGER.info(f"Sample in well {sample_doc[FIELD_COORDINATE]}")
+
+                # update sample in Mongo ‘samples’ to set lh_sample_uuid and updated_timestamp
+                try:
+                    query = SQL_MLWH_GET_SAMPLE_FOR_MONGO_ID % {MLWH_MONGODB_ID: sample_doc[FIELD_MONGODB_ID]}
+                    mlwh_sample = next(mysql_generator(config=config, query=query))
+
+                    if mlwh_sample[FIELD_LH_SAMPLE_UUID] is None:
+                        continue
+
+                    sample_doc[FIELD_LH_SAMPLE_UUID] = mlwh_sample[MLWH_LH_SAMPLE_UUID]
+                    sample_doc[FIELD_UPDATED_AT] = datetime.utcnow()
+
+                    success = update_mongo_sample(samples_collection, sample_doc)
+                    if success:
+                        counter_mongo_update_successes += 1
+                    else:
+                        counter_mongo_update_failures += 1
+
+                except Exception as e:
+                    counter_mongo_update_failures += 1
+                    LOGGER.critical("Failed to update sample in Mongo for mongo id " f"{sample_doc[FIELD_MONGODB_ID]}")
+                    LOGGER.exception(e)
+
+    LOGGER.info(f"Count of successful Mongo updates = {counter_mongo_update_successes}")
+    LOGGER.info(f"Count of failed Mongo updates = {counter_mongo_update_failures}")
+
+
+def update_mongo_sample(samples_collection: Collection, sample_doc: SampleDoc) -> bool:
+    """Updates a sample in the Mongo samples collection
+
+    Arguments:
+        config {Config} -- application config specifying database details
+        sample_doc {SampleDoc} -- the sample document whose fields should be updated
+
+    Returns:
+        bool -- whether the updates completed successfully
+    """
+    try:
+        mongo_sample = samples_collection.find_one_and_update(
+            filter={
+                FIELD_MONGODB_ID: sample_doc[FIELD_MONGODB_ID],
+            },
+            update={
+                "$set": {
+                    FIELD_LH_SAMPLE_UUID: sample_doc[FIELD_LH_SAMPLE_UUID],
+                    FIELD_UUID_UPDATED: True,
```

Comment:
Is this flag needed? Are we going to use it anywhere else?

Reply:
Ah, ok, if other samples where we updated the uuid have already this flag then is probably better to use it here as well to use the same convention.